geode-junit/src/main/java/org/apache/geode/security/UpdatableUserAuthInitialize.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.security;
+
+import java.util.Properties;
+
+import org.apache.geode.distributed.DistributedMember;
+
+/**
+ * this is used in conjunction with ExpirableSecurityManager. It will create a new set of
+ * credentials every time getCredentials are called, and they will always be authenticated
+ * and authorized by the ExpirableSecurityManager.
+ *
+ * make sure reset is called after each test to clean things up.
+ */
+public class UpdatableUserAuthInitialize implements AuthInitialize {
+  // use static field for ease of testing since there is only one instance of this in each VM
+  private static String user;
+
+  @Override
+  public Properties getCredentials(Properties securityProps, DistributedMember server,
+      boolean isPeer) throws AuthenticationFailedException {
+    Properties credentials = new Properties();
+    credentials.put("security-username", user);
+    credentials.put("security-password", user);
+    return credentials;
+  }
+
+  public static String getUser() {
+    return user;
+  }
+
+  public static void setUser(String user) {
+    UpdatableUserAuthInitialize.user = user;
```

Comment:
If this is going to be set by another thread than those reading the `user` value then `user` should be `volatile` or otherwise be made thread safe and synchronized.

Reply:
It's probably best to just change the `user` field to be of type AtomicReference<String>:
```
private static final AtomicReference<String> USER = new AtomicReference<>();
```